contracts/dao/voting/vesting-lp-vault/src/contract.rs

```diff
@@ -0,0 +1,328 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Binary, Decimal256, Deps, DepsMut, Env, Fraction, MessageInfo, Response, Uint128,
+};
+use cw2::set_contract_version;
+use cwd_interface::voting::{
+    BondingStatusResponse, TotalPowerAtHeightResponse, VotingPowerAtHeightResponse,
+};
+use cwd_interface::Admin;
+use serde::Serialize;
+
+use crate::state::{CONFIG, DAO};
+use neutron_oracle::voting_power::voting_power_from_lp_tokens;
+use neutron_vesting_lp_vault::{
+    error::{ContractError, ContractResult},
+    msg::{ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg},
+    types::Config,
+};
+use vesting_base::msg::{QueryMsg as VestingLpQueryMsg, QueryMsgHistorical};
+
+pub(crate) const CONTRACT_NAME: &str = "crates.io:neutron-vesting-lp-vault";
+pub(crate) const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> ContractResult<Response> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let owner = match msg.owner {
+        Admin::Address { addr } => deps.api.addr_validate(addr.as_str())?,
+        Admin::CoreModule {} => info.sender.clone(),
```

Comment:
This actually won't work since vaults are never initialized by the DAO. That feature is needed only for those contracts that are initialized by the DAO during DAO initialization.

Reply:
This is also a leftover from sketched lockdrop-vault. Should I remove this excess logic?

Reply:
My personal take is yeah, it should be removed both from here and the lockdrop contract. Let's discuss with the team though, maybe I miss smth

Reply:
what about the rest of the vaults? seems we need a uniform solution here, e.g. replace `Admin` with `String` in all the vaults in a separate PR
<img width="412" alt="image" src="https://user-images.githubusercontent.com/34917380/233314759-f900b181-6721-4a95-94e4-a417920e3e6a.png">

Reply:
Moved to #54